### PR TITLE
apply unary minus spacing directly after equals sign

### LINKED ISSUE #28180: [Bug]: mathtext should distinguish between unary and binary minus
### Bug summary

TeX inserts a (thin?) space after a binary minus, but not before a unary minus.  Typographically, I believe this does look nicer.
mathtext does not distinguish the two cases and always puts in a thin space.  It would be nice if it followed tex's behavior.

### Code for reproduction

```Python
from pylab import *
figtext(.2, .6, "$a=-b-c$")
figtext(.2, .5, "$a=-b-c$", usetex=True)
```

### Actual outcome

Figure_1
Compare the spacing before the "b" and the "c": they are different with usetex, but not with mathtext.

### Expected outcome

Remove the space in mathtext before the unary minus.

### Additional information

Probably involves reading the TeXbook to figure out what is TeX's logic for distinguishing unary and binary minus.
Tagging as good first issue *solely* because there should be no API design at all, but medium (-hard) difficulty because that'll require both reading the TeXbook and learning about mathtext's implementation.  However I would guess there's no "strong" technical blockers either.

### Operating system

any

### Matplotlib Version

3.10.0.dev137+g9387431ab6

### Matplotlib Backend

qtagg

### Python version

3.12

### Jupyter version

no

### Installation

git checkout

### PR DESCRIPTION
**Pull request by OdileVidrine**
## PR summary
Fixes a visual inconsistency where mathtext does not modify the spacing around a unary minus sign after an equals sign like it does for a unary minus sign after a closing parenthesis. Closes #28180.

## PR checklist
- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Comment by anntzer** [comment]
The fix looks plausible (although there's also the case of a leading minus at the beginning of a equation, perhaps?).  Did you check the TeXbook for reference?

File: lib/matplotlib/_mathtext.py [hunk 1]
```diff
@@ -2285,7 +2285,7 @@ def symbol(self, s: str, loc: int,
             if (self._in_subscript_or_superscript or (
                     c in self._binary_operators and (
                     len(s[:loc].split()) == 0 or prev_char == '{' or
-                    prev_char in self._left_delims))):
+                    prev_char in self._left_delims or prev_char == '='))):
```
Comment (oscargus) [inline lib/matplotlib/_mathtext.py:2288]: It should not just be =, but all relational operators.

Comment (OdileVidrine) [reply lib/matplotlib/_mathtext.py:2288]: good idea, I will change that

**Review by QuLogic — COMMENTED** [review]
There's a comment above the `if` that could be updated.

File: lib/matplotlib/_mathtext.py [hunk 2]
```diff
@@ -2285,7 +2285,8 @@ def symbol(self, s: str, loc: int,
             if (self._in_subscript_or_superscript or (
                     c in self._binary_operators and (
                     len(s[:loc].split()) == 0 or prev_char == '{' or
-                    prev_char in self._left_delims))):
+                    prev_char in self._left_delims or
+                    prev_char in self._relation_symbols))):
```
[the remark this answers is not in the document]
Comment (OdileVidrine) [reply lib/matplotlib/_mathtext.py]: I made the change you suggested and tested it.

**Review by QuLogic — APPROVED** [review]

**Review by ksunden — APPROVED** [review]

**Comment by ksunden** [comment]
@OdileVidrine Congrats on your first PR merged to Matplotlib! Hope to hear from you again!

**Comment by OdileVidrine** [comment]
Thank you, its exciting to contribute to such a large project.

**Pull request merged**